packages/amplify-graphql-model-transformer/rds-lambda/handler.ts

```diff
@@ -24,10 +24,37 @@ export const run = async (event): Promise<any> => {
     adapter = await getDBAdapter(config);
   }
   const debugMode = process.env.DEBUG_MODE === 'true';
-  const result = await adapter.executeRequest(event, debugMode);
-  return result;
+  try {
+    return await adapter.executeRequest(event, debugMode);
+  } catch (e) {
+    if (isRetryableError(e)) {
+      return await retryWithRefreshedCredentials(event, debugMode)
+    }
+    throw e;
+  }
 };
 
+const retryWithRefreshedCredentials = async (event, debugMode): Promise<any> => {
+  try {
+    const config = await getDBConfig();
+    adapter = await getDBAdapter(config);
+    return await adapter.executeRequest(event, debugMode);
+  } catch (err) {
+    adapter = null;
+    throw err;
+  }
+};
+
+const isRetryableError = (error): boolean => {
+  // https://www.postgresql.org/docs/current/errcodes-appendix.html
+  const postgresRetryableError = error.code === '28P01' && error.routine === 'auth_failed';
```

Comment:
From looking at the PostgreSQL docs, `routine` isn't part of the public interface for errors. It appears to be the name of the [function that reports the error](https://github.com/postgres/postgres/blob/master/src/backend/libpq/auth.c#L248). 

Are we comfortable relying on this to determine the error type? Is the `error.code` check maybe enough here?

Reply:
Good point. Looking at the docs I think using only `code` for Postgres and only `errno` for MySQL should be sufficient.